### PR TITLE
[Hub Generated] Review request for Microsoft.Maps to add Creator resource under version preview/2020-02-01-preview

### PR DESCRIPTION
**Pull request by ihvo**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/ihvo/azure-rest-api-specs/dev-maps-Microsoft.Maps-2020-02-01-preview...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for AME onboarding.
- [x] [Validation tools](https://aka.ms/swaggertools) have been run on swagger spec(s) updated and errors have all been fixed with this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, [please view the FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories.
*	adding/removing APIs.
* adding/removing properties.
* adding/removing API-version.
* adding a new service in Azure.

*Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.*

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ihvo** [comment]
Private Atlas resource will be renamed to Creator. This change adds Creator operations to replace existing Private Atlas operations. We will deprecate Private Atlas later. Both resource types exist in the same "preview" API version.

**Review by stack111 — COMMENTED** [review]
Left feedback offline - looks good, only questions for me are about current version and retirement of accounts/privateAtlases.

File: specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json [hunk 1]
```diff
@@ -679,6 +679,250 @@
...
+        ],
+        "description": "Get all Creator instances for an Azure Map Account",
+        "operationId": "Creators_ListByAccount",
+        "x-ms-examples": {
+          "ListPrivateAtlasByAccount": {
+            "$ref": "./examples/ListMapsCreatorsByAccount.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null,
```
Comment (majastrz) [inline specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json:696]: >null [](start = 26, length = 4)

Do you need paging for these resources? (It's ok to declare paging in the swagger even if the service doesn't support it currently. This rename is also a good opportunity to add this without breaking changes if you need to.)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side with a non-blocking comment. Please take a look.

**Comment by ruowan** [comment]
Hi, Pls fix linter error.

**Comment by ihvo** [comment]
Hi @ruowan, linter errors suggest that new APIs should use the standard response message. I have referenced types from common-types/resource-management/v1/types.json#/definitions/ErrorResponse which have the same structure as required in https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content.

So, expected response schema is compliant with guidelines. Is that a tool issue? Or am I missing something? 
Could you please advise?

**Comment by ruowan** [comment]
You could reference v2 types, v2 has fixed this issue. Here is a link https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json.

File: specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json (hunk 1 above)
Comment (ihvo) [reply specification/maps/resource-manager/Microsoft.Maps/preview/2020-02-01-preview/maps-management.json:696]: Probably don't need paging. Creators are expensive resource. We don't expect more than one per account.

**Comment by ihvo** [comment]
> You could reference v2 types, v2 has fixed this issue. Here is a link https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json.

Hi @ruowan, after referencing v2 type I've got PropertyRemoved failures. Is linter getting diff to the last commit or to master?

**Comment by ihvo** [comment]
hi @ruowan ,
I confirmed the false positive using diff tool 
https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi9tYXN0ZXIvc3BlY2lmaWNhdGlvbi9tYXBzL3Jlc291cmNlLW1hbmFnZXIvTWljcm9zb2Z0Lk1hcHMvcHJldmlldy8yMDIwLTAyLTAxLXByZXZpZXcvbWFwcy1tYW5hZ2VtZW50Lmpzb24iLCJ0YXJnZXRVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi82M2M0MDAxMWExMDdkMTFhM2NiYmM3NWYzMGE2NmNhMjkwMDM2MTg1L3NwZWNpZmljYXRpb24vbWFwcy9yZXNvdXJjZS1tYW5hZ2VyL01pY3Jvc29mdC5NYXBzL3ByZXZpZXcvMjAyMC0wMi0wMS1wcmV2aWV3L21hcHMtbWFuYWdlbWVudC5qc29uIn0=

I'm using v1/types.json on existing APIs, with no changes to them. I'm using v2/types.json only on added APis.

Could you please take a look today?

**Review by ruowan — APPROVED** [review]

**Comment by ruowan** [comment]
Thanks for your feedback. It's actually a false positive. We will fix our tools later. 

Could you pls change all v1 to v2?  The only difference between v1 and v2 is `/definitions/ErrorResponse`.  Your service is a preview version. If no customer impact, better to migrate it to v2.

**Comment by ihvo** [comment]
> Thanks for your feedback. It's actually a false positive. We will fix our tools later.
> 
> Could you pls change all v1 to v2? The only difference between v1 and v2 is `/definitions/ErrorResponse`. Your service is a preview version. If no customer impact, better to migrate it to v2. We allow this kind of breaking change.

Agree that it is necessary to migrate v1 to v2, but we should make a new API version. Current preview version is used by customers and there would be impact if I change the rest of APIs from v1 to v2.

**Comment by ihvo** [comment]
> Thanks for your feedback. It's actually a false positive. We will fix our tools later.
> 
> Could you pls change all v1 to v2? The only difference between v1 and v2 is `/definitions/ErrorResponse`. Your service is a preview version. If no customer impact, better to migrate it to v2. We allow this kind of breaking change.

Confirmed that v2 matches what our RP actually returns. Updated all references to v2 types.

@ruowan , could you please complete this PR?

**Pull request merged**